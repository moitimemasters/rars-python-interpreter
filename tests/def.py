def foo(y):
    def bar(x):
        return x
    return y + bar(1)


def rec_foo(x):
    return x + (0 if x == 0 else rec_foo(x - 1))


lmbda_foo = lambda x, y, z: (x + y + z) / 3

rec_foo(10) + foo(1) + lmbda_foo(1, 2, 3)
