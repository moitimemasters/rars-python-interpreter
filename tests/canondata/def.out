file: tests/def.py
started lexing
ended lexing
started parsing ast
ended parsing ast
started compiling
compiling node: 28
compiling node: 28
compiling node: 31
compiling node: 5
compiling node: 31
compiling node: 6
compiling node: 5
compiling node: 14
compiling node: 0
compiling node: 5
compiling node: 28
compiling node: 31
compiling node: 6
compiling node: 5
compiling node: 9
compiling node: 6
compiling node: 5
compiling node: 0
compiling node: 0
compiling node: 14
compiling node: 6
compiling node: 5
compiling node: 0
compiling node: 5
compiling node: 8
compiling node: 13
compiling node: 6
compiling node: 6
compiling node: 6
compiling node: 5
compiling node: 5
compiling node: 5
compiling node: 0
compiling node: 6
compiling node: 6
compiling node: 14
compiling node: 0
compiling node: 5
compiling node: 14
compiling node: 0
compiling node: 5
compiling node: 14
compiling node: 0
compiling node: 0
compiling node: 0
compiling node: 5
ended compiling
compiled instructions:
	 Instruction: 10
	 Instruction: 8
	 Instruction: 10
	 Instruction: 8
	 Instruction: 10
	 Instruction: 8
	 Instruction: 2
	 Instruction: 0
	 Instruction: 12
	 Instruction: 2
	 Instruction: 0
	 Instruction: 12
	 Instruction: 7
	 Instruction: 2
	 Instruction: 2
	 Instruction: 2
	 Instruction: 0
	 Instruction: 12
	 Instruction: 7
started interpreting
ended interpreting
Interpretation result: 59.0
